va/caa.go

```diff
@@ -229,30 +241,51 @@ func (va *ValidationAuthorityImpl) validateCAASet(caaSet *CAASet, wildcard bool)
 	//
 	// Our CAA identity must be found in the chosen checkSet.
 	for _, caa := range records {
-		if extractIssuerDomain(caa) == va.issuerDomain {
-			va.stats.Inc("CAA.Authorized", 1)
-			return true, true
+		caaIssuerDomain, caaParameters := extractIssuerDomainAndParameters(caa)
+		if caaIssuerDomain != va.issuerDomain {
+			continue
+		}
+
+		if features.Enabled(features.CAAValidationMethods) {
+			caaMethods, ok := caaParameters["validation-methods"]
+			if ok {
+				if challengeType == nil {
+					continue
+				}
+				if !containsMethod(caaMethods, *challengeType) {
+					continue
+				}
+			}
 		}
+
+		va.stats.Inc("CAA.Authorized", 1)
+		return true, true
 	}
 
 	// The list of authorized issuers is non-empty, but we are not in it. Fail.
 	va.stats.Inc("CAA.Unauthorized", 1)
 	return true, false
 }
 
+func isIssueSpace(r rune) bool {
```

Comment:
Since this is only used in `extractIssuerDomainAndParameters` could it be moved out of the package scope into the calling function's scope?

Reply:
Sure - done.